File: pinyougou/pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    public List<TbBrand> findAll();

    /**
     * 品牌分页
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    /**
     * 增加品牌
     * @param tbBrand
     */
    public void add(TbBrand tbBrand);

    /**
     * 根据主键查品牌
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改
     * @param brand
     */
    public void update(TbBrand brand);

    /**
     * 根据品牌id删除品牌
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 带条件品牌分页
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);

    /**
     * 返回下拉列表数据
     * @return
     */
    public List<Map> selectOptionList();
}

File: pinyougou/pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService", function ($http) {
    //查询所有
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }
    //分页
    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    }
    //根据id查询
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }
    //增加
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    }
    //更新
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    }
    //删除
    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    }
    //搜寻
    this.search = function (page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, searchEntity);
    }

    //下拉列表数据
    this.findBrandList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
});